addons/survey/models/survey.py

```diff
@@ -0,0 +1,1100 @@
+    user_input_line_ids = fields.One2many('survey.user_input_line',
+                                           'user_input_id', string='Answers', copy=True)
+
+    # URLs used to display the answers
+    result_url = fields.Char(related='survey_id.result_url',
+                                 string="Public link to the survey results")
+    print_url = fields.Char(related='survey_id.print_url',
+                                string="Public link to the empty survey")
+
+    quizz_score = fields.Float(compute="_compute_quizz_score", string="Score for the quiz")
+
+    _sql_constraints = [
+        ('unique_token', 'UNIQUE (token)', 'A token must be unique!'),
+        ('deadline_in_the_past', 'CHECK (deadline >= date_create)', 'The deadline cannot be in the past')
+    ]
+
+    @api.multi
+    def _compute_quizz_score(self):
+        result = self.env['survey.user_input_line'].read_group([('user_input_id', 'in', self.ids)], ['quizz_mark', 'user_input_id'], ['user_input_id'])
+        val = {res['user_input_id'][0]: res['quizz_mark'] for res in result}
+        for user_input in self:
+            user_input.quizz_score = val.get(user_input.id, 0.0)
+
+    @api.multi
+    def action_survey_resent(self):
+        ''' Sent again the invitation '''
+        self.ensure_one()
+        context = dict(self.env.context or {})
+        context.update({
+            'survey_resent_token': True,
+            'default_partner_ids': self.partner_id.id and [self.partner_id.id] or [],
+            'default_multi_email': self.email or "",
+            'default_public': 'email_private',
+        })
+        return self.survey_id.with_context(context).action_send_survey()
+
+    @api.multi
+    def action_view_answers(self):
+        ''' Open the website page with the survey form '''
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "View Answers",
+            'target': 'self',
+            'url': '%s/%s' % (self.print_url, self.token)
+        }
+
+    @api.multi
+    def action_survey_results(self):
+        ''' Open the website page with the survey results '''
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Survey Results",
+            'target': 'self',
+            'url': self.result_url
+        }
+
+    @api.one
+    def copy_data(self, default=None):
+        raise UserError(_('You cannot duplicate this element!'))
+
+    @api.model
+    def do_clean_emptys(self, automatic=False):
+        ''' Remove empty user inputs that have been created manually
+            (used as a cronjob declared in data/survey_cron.xml) '''
+        empty_user_inputs = self.search([('input_type', '=', 'manually'),
+                                                     ('state', '=', 'new'),
+                                                     ('date_create', '<', fields.Datetime.to_string((datetime.datetime.now() - datetime.timedelta(hours=1))))])
```

Comment:
whyuse python datetime function ? you unnecessary import datetime for this methods do it like :
```
fields.Datetime.to_string((fields.datetime.now() - relativedelta.relativedelta(hours=1)
```